src/test/java/fi/otavanopisto/kuntaapi/test/server/unit/ptv/PtvTranslatorTest.java

```diff
@@ -94,10 +99,37 @@ public void testMerged() throws IOException, JSONException {
     }
   }
   
-  private void assertServiceHoursEqual(List<ServiceHour> expected, List<ServiceHour> actual) throws IOException, JSONException {
-    String expectedString = getObjectMapper().writeValueAsString(expected);
-    String actualString = getObjectMapper().writeValueAsString(actual);
-    JSONAssert.assertEquals("ServiceHours are not equal", expectedString, actualString, false);
+  private void assertServiceHoursEqual(List<ServiceHour> expectedServiceHours, List<ServiceHour> actualServiceHours) throws IOException, JSONException {
+    assertEquals(expectedServiceHours.size(), actualServiceHours.size());
+    for (int i = 0; i < expectedServiceHours.size(); i++) {
+      ServiceHour expectedServiceHour = expectedServiceHours.get(i);
+      ServiceHour actualServiceHour = actualServiceHours.get(i);
+      assertEquals(expectedServiceHour.getServiceHourType(), actualServiceHour.getServiceHourType());
+      assertInstantsMatch(expectedServiceHour.getValidFrom(), actualServiceHour.getValidFrom());
+      assertInstantsMatch(expectedServiceHour.getValidTo(), actualServiceHour.getValidTo());
+      assertEquals(expectedServiceHour.getIsClosed(), actualServiceHour.getIsClosed());
+      assertEquals(expectedServiceHour.getValidForNow(), actualServiceHour.getValidForNow());
+      assertJSONEquals(expectedServiceHour.getAdditionalInformation(), actualServiceHour.getAdditionalInformation());
+      assertJSONEquals(expectedServiceHour.getOpeningHour(), actualServiceHour.getOpeningHour());
+    }
+  }
+
+  private void assertJSONEquals(Object expected, Object actual) throws JsonProcessingException, JSONException {
+    ObjectMapper objectMapper = new ObjectMapper();
+
+    JSONAssert.assertEquals(objectMapper.writeValueAsString(expected), objectMapper.writeValueAsString(actual), false);
+  }
+
+  private void assertInstantsMatch(OffsetDateTime expected, OffsetDateTime actual) {
+    if (expected == actual) {
+      return;  
+    }
+    
+    if (expected == null || actual == null) {
+      fail(String.format("ServiceHours (%s, %s) are not equal", expected, actual));
+    }
+    
+    assertTrue(String.format("ServiceHours (%s, %s) are not equal", expected, actual), sameInstant(expected.toInstant()).matches(actual.toInstant()));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "actual" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "expected" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)